src/main/java/org/wso2/carbon/identity/application/authenticator/magiclink/JWTGenerator.java

```diff
@@ -0,0 +1,71 @@
+package org.wso2.carbon.identity.application.authenticator.magiclink;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.PlainJWT;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * JWT Generator related to magic link authentication flow.
+ */
+
+@SuppressWarnings("checkstyle:LineLength")
+public class JWTGenerator {
+
+    private final JWSAlgorithm signatureAlgorithm = new JWSAlgorithm(JWSAlgorithm.RS256.getName());
+
+    /**
```

Comment:
Add a comment line mentioning the functionality of this method.
Update in other places also.